src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -5463,28 +5463,72 @@ void CalculateZoneMRT(EnergyPlusData &state,
                     surface.Area * state.dataConstruction->Construct(surface.Construction).InsideAbsorpThermal;
                 int ZoneNum = surface.Zone;
                 if (ZoneNum > 0) state.dataHeatBalSurfMgr->ZoneAESum(ZoneNum) += state.dataHeatBalSurfMgr->SurfaceAE(SurfNum);
+                if (surface.RadEnclIndex > 0)
+                    state.dataViewFactor->EnclRadInfo(surface.RadEnclIndex).sumAE += state.dataHeatBalSurfMgr->SurfaceAE(SurfNum);
             }
         }
     }
 
+    // Zero sumAET for applicable enclosures
+    if (present(ZoneToResimulate)) {
+        for (int spaceNum : state.dataHeatBal->Zone(ZoneToResimulate).spaceIndexes) {
+            int enclNum = state.dataHeatBal->space(spaceNum).radiantEnclosureNum;
+            state.dataViewFactor->EnclRadInfo(enclNum).sumAET = 0.0;
+            state.dataViewFactor->EnclRadInfo(enclNum).reCalcMRT = true;
+        }
+    } else {
+        for (auto &thisEnclosure : state.dataViewFactor->EnclRadInfo) {
+            thisEnclosure.sumAET = 0.0;
+            thisEnclosure.reCalcMRT = true;
+        }
+    }
     for (int ZoneNum = 1; ZoneNum <= state.dataGlobal->NumOfZones; ++ZoneNum) {
         if (present(ZoneToResimulate) && (ZoneNum != ZoneToResimulate)) continue;
-        Real64 SumAET = 0.0;
+        Real64 zoneSumAET = 0.0;
         for (int spaceNum : state.dataHeatBal->Zone(ZoneNum).spaceIndexes) {
             auto const &thisSpace = state.dataHeatBal->space(spaceNum);
             for (int SurfNum = thisSpace.HTSurfaceFirst; SurfNum <= thisSpace.HTSurfaceLast; ++SurfNum) {
-                SumAET += state.dataHeatBalSurfMgr->SurfaceAE(SurfNum) * state.dataHeatBalSurf->SurfTempIn(SurfNum);
+                Real64 surfAET = state.dataHeatBalSurfMgr->SurfaceAE(SurfNum) * state.dataHeatBalSurf->SurfTempIn(SurfNum);
+                zoneSumAET += surfAET;
+                state.dataViewFactor->EnclRadInfo(state.dataSurface->Surface(SurfNum).RadEnclIndex).sumAET += surfAET;
             }
-            if (state.dataHeatBalSurfMgr->ZoneAESum(ZoneNum) > 0.01) {
-                state.dataHeatBal->ZoneMRT(ZoneNum) = SumAET / state.dataHeatBalSurfMgr->ZoneAESum(ZoneNum);
+        }
+        if (state.dataHeatBalSurfMgr->ZoneAESum(ZoneNum) > 0.01) {
+            state.dataHeatBal->ZoneMRT(ZoneNum) = zoneSumAET / state.dataHeatBalSurfMgr->ZoneAESum(ZoneNum);
+        } else {
+            if (state.dataHeatBalSurfMgr->CalculateZoneMRTfirstTime) {
+                ShowWarningError(
+                    state,
+                    format("Zone areas*inside surface emissivities are summing to zero, for Zone=\"{}\"", state.dataHeatBal->Zone(ZoneNum).Name));
+                ShowContinueError(state, "As a result, MRT will be set to MAT for that zone");
+            }
+            state.dataHeatBal->ZoneMRT(ZoneNum) = state.dataZoneTempPredictorCorrector->zoneHeatBalance(ZoneNum).MAT;
+        }
```

Comment:
I am staring at this if/else and wondering what the for loop at lines 5488 - 5495 represents when ZoneAESum(ZoneNum) <= 0.01. Same for the enclosure loop below at 5511. If ZoneAESum(ZoneNum) <= 0.01, then isn't sumAE also <= 0.01? And doesn't that mean that the for loop could be moved into this if block above line 5497? The only draw-back I see is the calculation of state.dataViewFactor->EnclRadInfo(state.dataSurface->Surface(SurfNum).RadEnclIndex).sumAET would only occur when ZoneAESum > 0.01.

Reply:
@rraustad Good point - no sense in accumulating sumAET if it's not going to be used, because sumAE is zero.

Reply:
Excellent, thanks!